parts/k8s/kubernetesmasterresourcesvmss.t

```diff
@@ -0,0 +1,468 @@
+          "sourcePortRange": "*"
+        }
+      },
+    {{end}}
+      {
+        "name": "allow_ssh",
+        "properties": {
+          "access": "Allow",
+          "description": "Allow SSH traffic to master",
+          "destinationAddressPrefix": "*",
+          "destinationPortRange": "22-22",
+          "direction": "Inbound",
+          "priority": 101,
+          "protocol": "Tcp",
+          "sourceAddressPrefix": "*",
+          "sourcePortRange": "*"
+        }
+      },
+      {
+        "name": "allow_kube_tls",
+        "properties": {
+          "access": "Allow",
+          "description": "Allow kube-apiserver (tls) traffic to master",
+          "destinationAddressPrefix": "*",
+          "destinationPortRange":"443-443",
+          "direction": "Inbound",
+          "priority": 100,
+          "protocol": "Tcp",
+          "sourceAddressPrefix": "*",
+          "sourcePortRange": "*"
+        }
+      }
+    ]
+  },
+  "type": "Microsoft.Network/networkSecurityGroups"
+},
+{{if RequireRouteTable}}
+{
+  "apiVersion": "[variables('apiVersionDefault')]",
+  "location": "[variables('location')]",
+  "name": "[variables('routeTableName')]",
+  "type": "Microsoft.Network/routeTables"
+},
+{{end}}
+{{if not .MasterProfile.IsCustomVNET}}
+{
+  "apiVersion": "[variables('apiVersionDefault')]",
+  "dependsOn": [
+    {{if RequireRouteTable}}
+    "[concat('Microsoft.Network/routeTables/', variables('routeTableName'))]",
+    {{end}}
+    "[concat('Microsoft.Network/networkSecurityGroups/', variables('nsgName'))]"
+    ],
+    "location": "[variables('location')]",
+    "name": "[variables('virtualNetworkName')]",
+    "properties": {
+      "addressSpace": {
+        "addressPrefixes": [
+          "[parameters('vnetCidr')]"
+        ]
+      },
+      "subnets": [
+      {
+        "name": "subnetmaster",
+        "properties": {
+          "addressPrefix": "[parameters('masterSubnet')]"
+          ,"networkSecurityGroup": {
+            "id": "[variables('nsgID')]"
+          }
+          {{if RequireRouteTable}}
+          ,"routeTable": {
+            "id": "[variables('routeTableID')]"
+          }
+          {{end}}
+        }
+      },
+      {  
+        "name":"subnetagent",
+        "properties":{  
+            "addressPrefix": "[parameters('agentSubnet')]",
+            "networkSecurityGroup": {
+            "id": "[variables('nsgID')]"
+          }
+          {{if RequireRouteTable}}
+          ,"routeTable": {
+            "id": "[variables('routeTableID')]"
+          }
+          {{end}}
+        }
+      }
+      ]
+    },
+    "type": "Microsoft.Network/virtualNetworks"
+},
+{{end}}
+{
+  "apiVersion": "[variables('apiVersionDefault')]",
+  "location": "[variables('location')]",
+  "name": "[variables('masterPublicIPAddressName')]",
+  "properties": {
+    "dnsSettings": {
+      "domainNameLabel": "[variables('masterFqdnPrefix')]"
+    },
+    "publicIPAllocationMethod": "Dynamic"
```

Comment:
hi @ritazh Could you recall why this `publicIPAllocationMethod` value is  `Dynamic` in VMSS? We are implementing AKS whitelist feature which requires all master node should have static public IP, would like to know the reason why you set as `Dynamic` in VMSS at that time, thanks.
cc @feiskyer

Reply:
Hey @andyzhangx, I think `publicIPAllocationMethod` should be set to`Static`. At the time of this PR, I was trying to make sure `parts/k8s/kubernetesmasterresourcesvmss.t` resembled `parts/k8s/kubernetesmasterresources.t`. Looks like https://github.com/Azure/aks-engine/commit/7df12dd1ea5688bafe45511f40797511de3744a6 updated `publicIPAllocationMethod` to `Static` for `parts/k8s/kubernetesmasterresources.t`.  And https://github.com/Azure/acs-engine/pull/4101/files updated `publicIPAllocationMethod` to `Static` when using Standard LB for `parts/k8s/kubernetesmasterresourcesvmss.t`.

Reply:
thanks, let me update `publicIPAllocationMethod` to `Static` in vmss.